envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/EnvoyListenersFactory.kt

```diff
@@ -49,6 +55,56 @@ class EnvoyListenersFactory(
     private val egressRdsInitialFetchTimeout: Duration = durationInSeconds(20)
     private val ingressRdsInitialFetchTimeout: Duration = durationInSeconds(30)
 
+    private val tlsProperties = snapshotProperties.incomingPermissions.tlsAuthentication
+    private val requireClientCertificate = BoolValue.of(tlsProperties.requireClientCertificate)
+    private val minimumProtocolVersion = mapToProto(tlsProperties.protocol.minimumVersion)
+    private val maximumProtocolVersion = mapToProto(tlsProperties.protocol.maximumVersion)
+
+    private fun mapToProto(protocolVersion: String): TlsParameters.TlsProtocol {
+        return when (protocolVersion) {
+            "TLSv1_3" -> TlsParameters.TlsProtocol.TLSv1_3
+            "TLSv1_2" -> TlsParameters.TlsProtocol.TLSv1_2
+            "TLSv1_1" -> TlsParameters.TlsProtocol.TLSv1_1
+            "TLSv1_0" -> TlsParameters.TlsProtocol.TLSv1_0
+            "AUTO" -> TlsParameters.TlsProtocol.TLS_AUTO
```

Comment:
Why not use `TlsParameters.TlsProtocol` enum in properties directly? Does this mapping to proto give us any benefits?

We already use proto enum in properties - see a property `loadBalancing.policy` -> `Cluster.LbPolicy`

Reply:
I did not know we use that and spring would be able to convert that. @pbetkier said it would not that's why I created the mapping. Are you sure it works?

Reply:
I'm not 100% sure, but it works for `Cluster.LbPolicy`. I think it is easy to verify.